backend/wmg/pipeline/summary_cubes/expression_summary/load.py

```diff
@@ -57,8 +59,16 @@ def build_in_mem_cube(
         vals["sum"][idx : idx + n_vals] = cube_sum[cube_idx, mask]
         vals["nnz"][idx : idx + n_vals] = cube_nnz[cube_idx, mask]
 
+        dataset_index = 0
+
         for i, k in enumerate(other_cube_attrs):
-            vals[k][idx : idx + n_vals] = attr_values[i]
+            if k == "dataset_id":
+                dataset_index = i
+            if k != "publication_citation":
+                vals[k][idx : idx + n_vals] = attr_values[i]
+
+        dataset_dict = return_dataset_dict_w_publications()
```

Comment:
put this outside the for loop - we don't want it to run every iteration.

Reply:
Oh I thought line 68 needed to be run for each column name. And then everything from line 70 down runs outside of the for loop?

Reply:
I mean outside the outer for loop

Reply:
`dataset_dict = return_dataset_dict_w_publications()` We don't want this line to run for every row of the cube index